store/src/java-test/com/zimbra/cs/mailbox/RecurringTaskTest.java

```diff
@@ -0,0 +1,89 @@
+package com.zimbra.cs.mailbox;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.google.common.collect.Maps;
+import com.zimbra.common.account.Key;
+import com.zimbra.common.calendar.ZCalendar.ZCalendarBuilder;
+import com.zimbra.common.calendar.ZCalendar.ZVCalendar;
+import com.zimbra.common.mime.MimeConstants;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.cs.account.Account;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.mailbox.calendar.IcsImportParseHandler.ImportInviteVisitor;
+import com.zimbra.cs.mailbox.calendar.Invite;
+import com.zimbra.cs.mailbox.calendar.Recurrence;
+
+public class RecurringTaskTest {
+
+    @BeforeClass
+    public static void init() throws Exception {
+        MailboxTestUtil.initServer();
+
+        Provisioning prov = Provisioning.getInstance();
+        prov.createAccount("test@zimbra.com", "secret", Maps.<String, Object>newHashMap());
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        MailboxTestUtil.clearData();
+    }
+
+    @Test
+    public void testTaskRecurrenceDuration() throws ServiceException, UnsupportedEncodingException {
+        Account acct = Provisioning.getInstance().get(Key.AccountBy.name, "test@zimbra.com");
+        Mailbox mbox = MailboxManager.getInstance().getMailboxByAccount(acct);
+
+        String task = "BEGIN:VCALENDAR\n"
+                    + "PRODID:Zimbra-Calendar-Provider\n"
+                    + "VERSION:2.0\n"
+                    + "METHOD:PUBLISH\n"
+                    + "BEGIN:VTIMEZONE\n"
+                    + "TZID:Africa/Harare\n"
+                    + "BEGIN:STANDARD\n"
+                    + "DTSTART:16010101T000000\n"
+                    + "TZOFFSETTO:+0200\n"
+                    + "TZOFFSETFROM:+0200\n"
+                    + "TZNAME:CAT\n"
+                    + "END:STANDARD\n"
+                    + "END:VTIMEZONE\n"
+                    + "BEGIN:VTODO\n"
+                    + "UID:93077c29_ab51_41ad_aaa8_f63a68f963a6_migwiz\n"
+                    + "RRULE:FREQ=MONTHLY;INTERVAL=1;BYDAY=MO,TU,WE,TH,FR;BYSETPOS=5\n"
+                    + "SUMMARY:Pay Cassim Salary\n"
+                    + "ATTENDEE;CN=Salim Jhavary;CUTYPE=RESOURCE;ROLE=NON-PARTICIPANT;PARTSTAT=NEEDS-ACTION:mailto:salim@sjcasa.co.za\n"
```

Comment:
Please anonymize this data!  For instance, I assume this is a customer's email address - and github is a world readable system!!!!